cheroot/workers/threadpool.py

```diff
@@ -223,6 +234,7 @@ def shrink(self, amount):
         # to remove. As each request is processed by a worker, that worker
         # will terminate and be culled from the list.
         for n in range(n_to_remove):
+            self._pending_shutdowns.append(None)
```

Comment:
You probably wanted to use the constant here:
```suggestion
            self._pending_shutdowns.append(_SHUTDOWNREQUEST)
```

Reply:
No, the object we store in `_pending_shutdowns` is irrelevant. That constant is important in the requests queue.

Reply:
Maybe have another named constant for this? I don't like it being a magic value with no meaning.

Reply:
@the-allanc WDYT?

Reply:
It's not a magic value, it's a _meaningless_ value. Naming a constant for it and then not referring to that named constant anywhere else would be more confusing.

I needed something to act as a thread-safe counter, and using a deque seemed to be the most straight-forward option.